spi/core-spi/src/main/java/org/eclipse/dataspaceconnector/spi/types/TypeManager.java

```diff
@@ -16,53 +16,117 @@
 
 import com.fasterxml.jackson.core.type.TypeReference;
 import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonSerializer;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.SerializationFeature;
 import com.fasterxml.jackson.databind.jsontype.NamedType;
 import com.fasterxml.jackson.databind.module.SimpleModule;
 import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
 import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.jetbrains.annotations.NotNull;
 
 import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
 
 /**
  * Manages system types and is used to deserialize polymorphic types.
  */
 public class TypeManager {
     private final ObjectMapper objectMapper;
 
+    /**
+     * Concurrent support is not needed since this map is only populated a boot, which is single-threaded
+     */
+    private final Map<String, ObjectMapper> objectMappers = new HashMap<>();
+
+    /**
+     * Constructor without params.
+     */
     public TypeManager() {
         objectMapper = new ObjectMapper();
         objectMapper.registerModule(new JavaTimeModule()); // configure ISO 8601 time de/serialization
         objectMapper.configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false); // serialize dates in ISO 8601 format
         objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
-        SimpleModule module = new SimpleModule();
-        objectMapper.registerModule(module);
     }
 
+    /**
+     * Returns the object mapper for the default serialization context.
+     */
     public ObjectMapper getMapper() {
         return objectMapper;
     }
 
+    /**
+     * Returns the object mapper for the given serialization context, creating one based on the default mapper if required.
+     */
+    @NotNull
+    public ObjectMapper getMapper(String key) {
+        return objectMappers.computeIfAbsent(key, k -> objectMapper.copy());
+    }
+
+    /**
+     * Registers types with all contexts.
+     */
     public void registerTypes(Class<?>... type) {
         objectMapper.registerSubtypes(type);
```

Comment:
What do you think about adding the `objectMapper` to the `objectMappers`, for example using "`default`" as key? that would make line 72 and 80 obsolete.